pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -559,6 +567,58 @@ private BrokerResponseNative handleSQLRequest(long requestId, String query, Json
     return brokerResponse;
   }
 
+  private void handleTimestampIndexOverride(PinotQuery pinotQuery)
+      throws JsonProcessingException {
+    Map<Integer, String> pushdownExpressions = new HashMap<>();
+    for (Expression expression : pinotQuery.getSelectList()) {
+      getPushdownExpression(expression, pushdownExpressions);
+      if (!pushdownExpressions.isEmpty()) {
+        pinotQuery.getQueryOptions()
+            .put("pushdown.timestamp.index.expression", new ObjectMapper().writeValueAsString(pushdownExpressions));
```

Comment:
Constructing an ObjectMapper is extremely expensive, can you make this a static or instance member instead? Since they contain two interning tables it’s generally better to share one of these between the entire application, unless you need different settings.

Reply:
Yes, this is for my testing, trying to set the override expressions as a map in the query option, then override the function to generated column names.

This should be simpler than modifying each individual transform function. Also, we can leverage a similar mechanism for function indexing later on. (Timestamp index can be treated as a special function index.)